### PR TITLE
This pull request is for reviewing changes on Team 2 task4 automation script.

### PR DESCRIPTION
**Pull request by ahmadafrasiyab**

### CODE REVIEW

**Comment by ahmadafrasiyab** [comment]
### Time tracking:
| **Date** | **Activity** | **Time Spent** |
|----------|-------------|-----------------|
| 10 Jan | Setting up issue | 15 minutes |
| 11 Jan | Researching Templates | 3 hours |
| 12 Jan | Setting up visual studio code environment | 1 hour |
| 13 Jan | Editing Template | 45 minutes |
| 14 Jan | Troubleshooting Issues | 8 hours |

**Comment by ahmadafrasiyab** [comment]
Steps to Run the Script:
1. Pull the code on your local machine.
2. When inside the repo, navigate to Automation\Task4
3. The script can be run using the following command: 
./deploy_template_task4 -SubscriptionId [Enter Sub Id] -TenantId 39edf504-8b38-44d6-bc9d-992fc9bb30ed -ServicePrincipalId ad1f8e28-7c7d-403c-a317-b2dd02c39cc0 -ServicePrincipalPassword MLT3O4wom9m7Q6dCU~~Vh0Q9.o5_yil7Rl -Resource_Group_Name afrasasas -function_name myfunction221  -storage_account_name mystorage221 -app_service_plan_name myappservice221 -location westus2 -template_file_path ./template.json

**Comment by yergu** [comment]
I  was trying to run your rescript in my local machine and I was not able to login to Azure , I kept on getting the similar error message   " ClientSecretCredential authentication failed" as you mentioned on slack and a warning message "$ServicePrincipalPassword' should use SecureString, otherwise this will expose sensitive information. "

File: Automation/Task4/deploy_template_task4.ps1 [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+Clear-AzContext -Force;
+$securePassword = ConvertTo-SecureString -String $ServicePrincipalPassword -AsPlainText -Force;
+$credentials = New-Object -TypeName System.Management.Automation.PSCredential($ServicePrincipalId, $securePassword);
+
+
+Connect-AzAccount -Credential $credentials -ServicePrincipal -Tenant $TenantId -SubscriptionId $SubscriptionId;
+
+New-AzResourceGroup -Name $Resource_Group_Name -Location $location
+
+New-AzResourceGroupDeployment -ResourceGroupName $Resource_Group_Name -TemplateFile "./template.json" -function_name $function_name -storage_account_name $storage_account_name -app_service_plan_name $app_service_plan_name -location $location
```
Comment (toddysm) [inline Automation/Task4/deploy_template_task4.ps1:57]: You can improve this by collecting the parameters in a HashTable. And you can pass the HashTable as parameter to New-AzResourceGtoupDeployment

File: Automation/Task4/deploy_template_task4.ps1 [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    [string]
+    $ServicePrincipalId,
+
+    [Parameter(Mandatory=$True)]
+    [String]
+    $ServicePrincipalPassword,
+
+    [Parameter(Mandatory=$True)]
+    [string]
+    $Resource_Group_Name,
```
Comment (toddysm) [inline Automation/Task4/deploy_template_task4.ps1:21]: Be consistent with the variable naming.

File: Automation/Task4/deploy_template_task4.ps1 (hunk 1 above)
Comment (ahmadafrasiyab) [reply Automation/Task4/deploy_template_task4.ps1]: I have so far been unable to fix the script with the hashtable. However, i have fixed the naming consistency issue. If you are ok, then we can merge this one and keep the hashtable as a future improvement in one of the next sprints.

File: Automation/Task4/deploy_template_task4.ps1 (hunk 2 above)
Comment (ahmadafrasiyab) [reply Automation/Task4/deploy_template_task4.ps1]: This has been fixed.

**Comment by ahmadafrasiyab** [comment]
This is just awaiting reviews now @jinudean @theanticrumpet @yergu Thanks.

**Comment by ahmadafrasiyab** [comment]
@toddysm I am closing this one without merge as pull#82 from @jinudean is much better, has logging and deploys without the one error that I am having.

**Comment by ahmadafrasiyab** [comment]
reopening for approval and merge

**Comment by yergu** [comment]
I am able to deployed your script successfully  and it works fine I verified the resources in the azure portal so you can merge it

**Review by yergu — APPROVED** [review]

**Comment by jinudean** [comment]
Good job

**Review by jinudean — APPROVED** [review]
I approved

**Pull request merged**